alibuild_helpers/build.py

```diff
@@ -61,26 +62,29 @@ def update_git_repos(args, specs, buildOrder, develPkgs):
     """
 
     def update_repo(package, git_prompt):
+        specs[package]["scm"] = Git()
+        if package in develPkgs:
+          localCheckout = os.path.join(os.getcwd(), specs[package]["package"])
+          if exists("%s/.sl" % localCheckout):
+            specs[package]["scm"] = Sapling()
```

Comment:
Is it worth extracting this into an e.g. `SCM.detect(dir)` function? The logic is repeated slightly differently elsewhere.